background.js

```diff
@@ -1,49 +1,53 @@
-browser.pageAction.onClicked.addListener(async () => {
-  await browser.tabs.executeScript({
-    code: `document.getElementsByClassName("download")[0].click();`,
-  });
-
-  await new Promise(r => setTimeout(r, 500));
+const clickElement = async querySelector => {
+  await browser.tabs.executeScript({code: `
+    if (document.querySelector("${querySelector}"))
+      document.querySelector("${querySelector}").click();
+  `});
+}
+
+const waitForElement = async querySelector => {
+  let [isElement] = await browser.tabs.executeScript({code: `document.querySelector("${querySelector}") !== null;`});
+  let timeLeft = 1000;
+
+  while (!isElement && timeLeft > 0) {
+    await new Promise(r => setTimeout(r, 100));
+    [isElement] = await browser.tabs.executeScript({code: `document.querySelector("${querySelector}") !== null;`});
+    timeLeft -= 100;
+  }
+
+  return isElement;
+}
+
+const waitAndClick = async querySelector => {
+  if (await waitForElement(querySelector))
+    await clickElement(querySelector);
+}
 
-  await browser.tabs.executeScript({
-    code: `document.getElementsByClassName("board-tab-item-underlined-component share-menu-tab-selector-tab")[0].click();`,
-  });
+browser.pageAction.onClicked.addListener(async () => {
+  await waitAndClick(".download");
 
-  await browser.tabs.executeScript({
-    code: `e = document.querySelector("[name='isComputerAnalysisEnabled']"); if (e !== null) e.click();`,
-  });
+  await waitAndClick(".board-tab-item-underlined-component");
 
-  setTimeout(async () => {
-    const [pgn] = await browser.tabs.executeScript({
-      code: `document.querySelector("[name='pgn']").value;`,
-    });
-    
-    await browser.tabs.executeScript({
-      code: `document.getElementsByClassName("icon-font-chess x ui_outside-close-icon")[0].click();`,
-    });
+  if (await waitForElement("[name='pgn']")) {
+    const [pgn] = await browser.tabs.executeScript({code: `document.querySelector("[name='pgn']").value;`});
 
-    await browser.tabs.create({ url: "https://lichess.org/paste" });
+    await waitAndClick("[data-cy='share-menu-close']");
 
-    await new Promise(r => setTimeout(r, 500));
+    await browser.tabs.create({url: "https://lichess.org/paste"});
 
-    const [logged] = await browser.tabs.executeScript({
-      code: `!document.getElementById("form3-analyse").disabled;`
-    });
+    if (await waitForElement("[name='analyse']")) {
+      const [logged] = await browser.tabs.executeScript({code: `!document.querySelector("[name='analyse']").disabled;`});
```

Comment:
The word _logged_ is more connected to _logging_ (storing log messages from a run of the program, for example to the console) than authentication. I assume this is meant to be connected to authentication. In that case, I suggest using the name `loggedIn` or `signedIn`, which is a more standard name related to auth :smile: 

```suggestion
      const [loggedIn] = await browser.tabs.executeScript({code: `!document.querySelector("[name='analyse']").disabled;`});
```